optuna/integration/lightgbm_tuner/optimize.py

```diff
@@ -302,8 +313,9 @@ def __init__(
         self.best_params = {} if best_params is None else best_params
         self.tuning_history = [] if tuning_history is None else tuning_history
 
-        if early_stopping_rounds is None:
-            self._suggest_early_stopping_rounds()
+        # Set default parameters as best.
+        self.best_params.update(DEFAULT_LIGHTGBM_PARAMETERS)
```

Comment:
This overwrites the `best_params` even when the user input a non-default value to the `__init__` argument. Is it expected?

Reply:
It is expected. I don't think this interface is easy to use, but it is the same design as the LightLGB's `evals_result`. `best_params` is used to store the best parameters.